traversal/planner/GraphPlanner.java

```diff
@@ -291,11 +295,31 @@ private void optimise(long timeLimitMillis) {
         isOptimising.set(false);
     }
 
+    private void linearise() {
+        Set<PlannerVertex<?>> visited = new HashSet<>();
+        LinkedList<PlannerVertex<?>> toVisit = vertices.values().stream().filter(PlannerVertex::isStartingVertex).collect(Collectors.toCollection(LinkedList::new));
```

Comment:
this can also be done with our iterators ;)

Reply:
It can now :)